Upcoming incompatibility with CPython 3.15

https://github.com/python/cpython/pull/141108 that removes an enum value https://github.com/python/cpython/pull/141108/files#diff-1541528aa8ba238fa13865ed284b8b100b9f35bfea78237cfd2f5506715d8367L27

```
g++ -fPIC -I/home/tcaswell/.virtualenvs/bisect/include -I/home/tcaswell/.pybuild/bisect/include/python3.15 -c src/greenlet/
greenlet.cpp -o build/temp.linux-x86_64-cpython-315/src/greenlet/greenlet.o
In file included from /home/tcaswell/.pybuild/bisect/include/python3.15/internal/pycore_backoff.h:12,
                 from /home/tcaswell/.pybuild/bisect/include/python3.15/internal/pycore_code.h:11,
                 from /home/tcaswell/.pybuild/bisect/include/python3.15/internal/pycore_interpframe.h:8,
                 from src/greenlet/TGreenlet.hpp:35,
                 from src/greenlet/greenlet_internal.hpp:18,
                 from src/greenlet/greenlet.cpp:19:
src/greenlet/TGreenlet.cpp: In member function ‘void greenlet::Greenlet::expose_frames()’:
src/greenlet/TGreenlet.cpp:636:41: error: ‘FRAME_OWNED_BY_CSTACK’ was not declared in this scope; did you mean ‘FRAME_OWNED_BY_THREAD’?
  636 |             assert(iframe_copy.owner != FRAME_OWNED_BY_CSTACK);
      |   
```

If `-DNDEBUG` is passed  then greenlet compiles with only warnings.  I had `-fpermissive` set in my `$CXXFLAGS` which seems to cause something in the build chain to drop `-DNDEBUG` .  There may be a better way to trigger this, but it setting `CXXFLAGS='-fpermissive'` is key to reproducing (it may be setting anything is enough?). 

Given that the assert is checking the variable has a value that no longer exists I think it should be safe to delete the line.

--- a/src/greenlet/TGreenlet.cpp
+++ b/src/greenlet/TGreenlet.cpp
@@ -633,7 +633,13 @@ void GREENLET_NOINLINE(Greenlet::expose_frames)()
             // directly.  This is important since GetFrameObject might
             // lazily _create_ the frame object and we don't want the
             // interpreter to lose track of it.
+            //
+            #if !GREENLET_PY315
+            // This enum value was removed in
+            //    https://github.com/python/cpython/pull/141108
+
             assert(iframe_copy.owner != FRAME_OWNED_BY_CSTACK);
+            #endif
 
             // We really want to just write:
             //     PyFrameObject* frame = _PyFrame_GetFrameObject(iframe);

--- a/src/greenlet/greenlet_cpython_compat.hpp
+++ b/src/greenlet/greenlet_cpython_compat.hpp
@@ -67,6 +67,12 @@ https://bugs.python.org/issue46090). Summary of breaking internal changes:
 #    define GREENLET_PY314 0
 #endif
 
+#if PY_VERSION_HEX >= 0x30F0000
+#    define GREENLET_PY315 1
+#else
+#    define GREENLET_PY315 0
+#endif
+
 #ifndef Py_SET_REFCNT
 /* Py_REFCNT and Py_SIZE macros are converted to functions
 https://bugs.python.org/issue39573 */

